src/adapters/kafka/index.ts

```diff
@@ -0,0 +1,99 @@
+import { Kafka } from 'kafkajs'
+import Adapter from '../../lib/adapter.js'
+import GleeMessage from '../../lib/message.js'
+
+class KafkaAdapter extends Adapter {
+  private kafka: Kafka
+  private firstConnect: boolean = true
+  name(): string {
+    return 'Kafka adapter'
+  }
+
+  async connect() {
+    const kafkaOptions = await this.resolveProtocolConfig('kafka')
+    const securityRequirements = (this.AsyncAPIServer.security() || []).map(
+      (sec) => {
+        const secName = Object.keys(sec.json())[0]
+        return this.parsedAsyncAPI.components().securityScheme(secName)
+      }
+    )
+    const userAndPasswordSecurityReq = securityRequirements.find(
+      (sec) => sec.type() === 'userPassword'
+    )
+    const scramSha256SecurityReq = securityRequirements.find(
+      (sec) => sec.type() === 'scramSha256'
+    )
+    const scramSha512SecurityReq = securityRequirements.find(
+      (sec) => sec.type() === 'scramSha512'
+    )
+  
+    const brokerUrl = new URL(this.AsyncAPIServer.url())
+    this.kafka = new Kafka({
+      clientId: 'glee-app', 
+      brokers: [brokerUrl.host],
+      ssl: {
+        rejectUnauthorized: kafkaOptions?.authentication?.rejectUnauthorized,
+        key: kafkaOptions?.authentication?.keyFilePath,
+        cert: kafkaOptions?.authentication?.cert,
+      },
+      sasl: {
+        mechanism: (scramSha256SecurityReq? kafkaOptions?.authentication?.scramSha256?.mechanism : undefined) || 
+                   (scramSha512SecurityReq? kafkaOptions?.authentication?.scramSha512?.mechanism : undefined),
+        username: userAndPasswordSecurityReq? kafkaOptions?.authentication?.userPassword?.username : undefined,
+        password: userAndPasswordSecurityReq? kafkaOptions?.authentication?.userPassword?.password : undefined,
```

Comment:
```suggestion
        mechanism: (scramSha256SecurityReq? 'scram-sha-256 : undefined) || 
          (scramSha512SecurityReq? 'scram-sha-512' : undefined) ||
          'plain',
        username: kafkaOptions?.authentication?.username,
        password: kafkaOptions?.authentication?.password,
```

A few things here:

1. If `scramSha256SecurityReq` is not `undefined`, it means it has to be used. No need to go to the config file. The AsyncAPI is the config file in this case. Same for `scramSha512SecurityReq`.
2. If another mechanism is specified, the `plain` should be used.
3. All mechanisms let you pass a username and password so let's not make it dependent on the security requirement.

Reply:
```
mechanism: (scramSha256SecurityReq? 'scram-sha-256 : undefined) || 
          (scramSha512SecurityReq? 'scram-sha-512' : undefined) ||
          'plain', 
```
          but this is giving an error saying : Type '{ mechanism: "plain" | "scram-sha-256" | "scram-sha-512"; username: any; password: any; }' is not assignable to type 'SASLOptions | Mechanism'.
  Object literal may only specify known properties, and 'username' does not exist in type 'Mechanism'.ts(2322)
index.d.ts(53, 3): The expected type comes from property 'sasl' which is declared here on type 'KafkaConfig

Reply:
Pushed a fix for this :)